### DISCUSSION TITLE
Previewing unsaved changes to a revision

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Revisions#910] [Feature Requests]
Custom report:

> I've noticed there's a bug with PublishPress Revision Pro. When I am working on updating content to a page and click the preview button, the update does not appear in the preview button. 

To be honest, I hadn't noticed that this was the default "Preview" behavior in WordPress, but I've tested and it is.

**Comment by stevejburge** [comment]
This seems like a useful new feature, rather than a bug.

From @agapetry 

> It's not a simple "bug" but a development task to meet expectations for a different type of preview.  I know that the WordPress editor convention is that preview means showing unsaved changes, but for Revisionary it has always meant showing revisions that are saved but unapproved or unpublished.  

> I can make previews from the revision editor use the WP autosave mechanism, but that actually becomes a UI ambiguity issue because in many cases there is no need for that slow autosave operation, just a front end view of what's already saved.

> I can understand why it will seem like a bug to many users.  Maybe for now we should change the button caption to "View" and "View / Approve"
 
> Another short-term option would be to disable that button if the post content is changed.  Then if the revision is saved, the button is re-enabled and again gives the same result as a true "preview" (edited)

**Comment by JoshJourney** [comment]
Indeed would be good to have the preview button save a temporary draft and show it on the frontend in a new tab. Temporary because it doesn't need to be stored longer than 15 minutes (especially large articles), new tab for convenience and so the currently edited revision doesn't get tampered by the preview.
